Build failure with GHC 8.6.5

As seen in [this CI job](https://github.com/GaloisInc/argo/actions/runs/6388041291/job/17337248567?pr=201):

```
[29 of 34] Compiling Network.Wai.Handler.Warp.HTTP2 ( Network/Wai/Handler/Warp/HTTP2.hs, dist/build/Network/Wai/Handler/Warp/HTTP2.o )

Network/Wai/Handler/Warp/HTTP2.hs:48:16: error:
Error:     • Constructor ‘H2.Config’ does not have the required strict field(s): confMySockAddr,
                                                                          confPeerSockAddr
    • In the expression:
        H2.Config
          {confWriteBuffer = bufBuffer writeBuffer,
           confBufferSize = bufSize writeBuffer, confSendAll = sendBS,
           confReadN = recvN, confPositionReadMaker = pReadMaker ii,
           confTimeoutManager = timeoutManager ii}
      In an equation for ‘conf’:
          conf
            = H2.Config
                {confWriteBuffer = bufBuffer writeBuffer,
                 confBufferSize = bufSize writeBuffer, confSendAll = sendBS,
                 confReadN = recvN, confPositionReadMaker = pReadMaker ii,
                 confTimeoutManager = timeoutManager ii}
      In the expression:
        do istatus <- newIORef False
           rawRecvN <- makeRecvN bs $ connRecv conn
           writeBuffer <- readIORef $ connWriteBuffer conn
           let recvN
                 = wrappedRecvN
                     th istatus (S.settingsSlowlorisSize settings) rawRecvN
               sendBS x = connSendAll conn x >> T.tickle th
               ....
           ....
   |
48 |         conf = H2.Config {
   |                ^^^^^^^^^^^...
```

Strictly speaking, the fault lies in an old Hackage release of the `warp` library, which doesn't version bounds correctly. While I could try to coordinate with the `warp` maintainers to revise the version bounds, I wonder if it wouldn't be more direct just to drop GHC 8.6 support in `argo`. We're not actively using it anymore, after all.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ghc-ver: ["8.6.5", "8.8.4", "8.10.7", "9.0.2", "9.2.8", "9.4.5", "9.6.2"]
+        ghc-ver: ["8.8.4", "8.10.7", "9.0.2", "9.2.8", "9.4.5", "9.6.2"]
         cabal-ver: ["3.10.1.0"]
       # complete all jobs
       fail-fast: false

--- a/README.rst
+++ b/README.rst
@@ -51,14 +51,14 @@ Build tools
 Requirements:
 
 * cabal-install 2.4.1.0 or newer
-* GHC-8.6.5 or GHC-8.8.4
+* GHC-8.8.4 or newer
 * Python 3.7 or higher
 
 Any easy to way get GHC and cabal-install installed is to use `ghcup`_;
 however any other method will be fine::
 
-    $ ghcup install 8.6.5
-    $ ghcup set 8.6.5 # optional
+    $ ghcup install 9.2.8
+    $ ghcup set 9.2.8 # optional
     $ ghcup install-cabal
 
 .. _ghcup: https://gitlab.haskell.org/haskell/ghcup

--- a/argo/argo.cabal
+++ b/argo/argo.cabal
@@ -31,7 +31,7 @@ common warnings
 
 common deps
   build-depends:
-    base                         >= 4.11.1.0 && < 4.19,
+    base                         >= 4.13 && < 4.19,
     aeson                        >= 1.4.2 && < 2.3,
     async                       ^>= 2.2,
     bytestring                   >= 0.10.8 && < 0.12,

